runtime/lua/vim/lsp/rpc.lua

```diff
@@ -278,10 +278,8 @@ function Client:encode_and_send(payload)
   if self.transport:is_closing() then
     return false
   end
-  local jsonstr = assert(
-    vim.json.encode(payload),
-    string.format("Couldn't encode payload '%s'", vim.inspect(payload))
-  )
+  local jsonstr = vim.json.encode(payload)
```

Comment:
Could you have added an `if` condition after this to maintain the same behaviour?

Reply:
If that's restored, the message should have a maximum (truncate). We should not print GB of data in messages/logs.